src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -207,7 +207,7 @@ internal bool TryWrite(Utf8JsonWriter writer, T value, JsonSerializerOptions opt
         {
             if (writer.CurrentDepth >= options.EffectiveMaxDepth)
             {
-                ThrowHelper.ThrowJsonException_SerializerCycleDetected(options.MaxDepth);
+                ThrowHelper.ThrowJsonException_SerializerCycleDetected(options.EffectiveMaxDepth);
```

Comment:
@thomaslevesque - are up for submitting a PR with tests for this? If not, @Jozkee - can you do so (or file an up-for-grabs issue)?

Reply:
I can try, if I figure out how to run the tests...
So far I've been unable to load the solution in VS. Do I need a preview version?

Reply:
@ViktorHofer can you please help @thomaslevesque here or direct him to the latest docs? Tyvm!

Reply:
@thomaslevesque what version do you have? Our projects require VS2019 (16.x) to open. Do you have 16.5.0 (see Help>About)? Although, earlier releases of 16.x may work.

Reply:
@ahsonkhan  @danmosemsft I eventually managed to build and run the tests. I just modified an existing test, I hope that's ok. See #34528 
It only tests one of the paths. As mentioned by @Jozkee, the other path seems unreachable.